docs/3.x/extend/graphql.md

```diff
@@ -489,6 +507,8 @@ class Widget extends craft\gql\base\ElementResolver
 
 If this wasn’t a custom element type, we’d need a class to extend <craft3:craft\gql\base\Resolver> instead, implementing a [resolve()](craft3:craft\gql\base\Resolver::resolve()) method to return the field’s value rather than translating it into an element query.
 
+The handling of `$source` here is important, because `prepareQuery()` may be called at different points in the request flow depending on exactly where `WidgetField` is accessed. `$source` will only have a `null` value if `WidgetField` is accessed at the top level of the query, and once populated `$fieldName` will either contain an array of relation IDs (if eager loading) or an array of already-fetched Widget objects.
+
```

Comment:
If eager-loading, `$fieldName` will contain an array of Widget objects.

Otherwise, it will contain a prepped element query object that, if executed, will return the Widgets from the DB.

Reply:
You are the wind beneath my derpy wings. Thank you.

Reply:
[That](https://github.com/craftcms/docs/commit/d3a8f80cf2fcbcadcb83b75490a010565d99ee4b) should do it!